lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -91,19 +90,20 @@ def __init__(self, config: PbenchServerConfig, logger: Logger, *schemas: ApiSche
         super().__init__(config, logger, *schemas)
 
         api_name = self.__class__.__name__
+        self.method = schemas[0].method
```

Comment:
We don't have any `ElasticBase` APIs that have two schemas, supporting two different HTTP methods, and I'm not entirely sure we'll need to do that... so I'm just pointing out here that your code won't support that.

I don't think it's worth complicating the code in this PR to handle that elegantly, however it might be a good idea to add a check for `len(schemas) > 1` and fail the constructor as done below on line 95 if no schema is supplied at all. (In fact, if that condition ever actually occurred, you'd be throwing an exception at line 93, so reversing these might not be a bad idea.)

(While `MissingDatasetNameParameter` has apparently already been somewhat twisted by this "no schema provided", one option here would be to simply change the `if not schemas` to `if len(schemas) != 1` with a slight tweak to the text.)